src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php

```diff
@@ -307,14 +309,21 @@ protected function createContainer(array $data = array())
 
     protected function createContainerFromFile($file, $data = array())
     {
-        $container = $this->createContainer($data);
-        $container->registerExtension(new FrameworkExtension());
-        $this->loadFromFile($container, $file);
+        $cacheKey = md5($file.serialize($data));
+
+        if (!isset(self::$containerCache[$cacheKey])) {
+            $container = $this->createContainer($data);
+
+            $container->registerExtension(new FrameworkExtension());
+            $this->loadFromFile($container, $file);
 
-        $container->getCompilerPassConfig()->setOptimizationPasses(array());
-        $container->getCompilerPassConfig()->setRemovingPasses(array());
-        $container->compile();
+            $container->getCompilerPassConfig()->setOptimizationPasses(array());
+            $container->getCompilerPassConfig()->setRemovingPasses(array());
+            $container->compile();
+
+            self::$containerCache[$cacheKey] = $container;
+        }
 
-        return $container;
+        return self::$containerCache[$cacheKey];
     }
```

Comment:
To minimize the diff, can you first check if the cache is present and immediately return the cache value?